Voice/Weather Phone IVR - Node.js/index.js

```diff
@@ -0,0 +1,184 @@
+        max: 1,
+        digitTimeout: 15,
+      },
+    });
+    const cmdResult = await cmdPrompt.waitForResult();
+    type = cmdResult.type;
+    digits = cmdResult.digits;
+    terminator = cmdResult.terminator;
+    console.log(
+      "Prompted for digits, received digits",
+      type,
+      digits,
+      terminator
+    );
+
+    if (digits === "1") {
+      // User input 1.  We are going to dial a Washington weather
+      // number and connect the call.
+      await call.connectPhone({
+        from: call.from,
+        to: DC_WEATHER_PHONE,
+        timeout: 30,
+        ringback: new Voice.Playlist().add(
+          Voice.Playlist.TTS({
+            text: "ring. ring. ring. ring. ring. ring. ring. ring",
+          })
+        ),
+      });
+      console.log("Connecting to DC weather phone ...");
+      await call.waitUntilConnected();
+      console.log("Connected");
+    } else if (digits === "2") {
+      // User input 2.  We are going to query a weather API, find the weather of Washington,
+      // and message that weather to the user's number.
+      const place = "Washington DC";
+      console.log(`Sending message about weather of ${place}`);
+      const weather = await getWeatherFromOpenWeatherMap(place);
+      const message = `${place} weather: ${
+        weather.weather[0].description
+      }. Temperature: ${(weather.main.temp - 273).toFixed(2)}°C`;
+      console.log(message, "being sent to number", call.to);
+      try {
+        await messageClient.send({
+          from: PHONE_NUMBER,
+          to: call.to,
+          body: message,
+        });
+      } catch (e) {
+        const pb = await call.playTTS({
+          text:
+            "Sorry, I couldn't send the message." +
+            (e?.data?.from_number[0] ?? " ") +
+            " I will say the contents here. " +
+            message,
+        });
+        await pb.waitForEnded();
+      }
+    } else if (digits === "3") {
+      //User input 3.  We are going to play a rain dance song hosted on our servers.
+      console.log("Sending rain dance song");
+      const rainDance = await call.playAudio({
+        url: "https://swrooms.com/rain.mp3",
+      });
+      await rainDance.waitForEnded();
+    } else if (digits === "4") {
+      //User input 4.  We are going to ask for a phone number to dial and play a rain dance song to it.
+      console.log("Sending rain song to your friend");
+      const prompt = await call.promptTTS({
+        text: "Please enter your friend's number then dial #. Please use the international format, but skip the plus sign.",
+        digits: {
+          max: 15,
+          digitTimeout: 15,
+          terminators: "#",
+        },
+      });
+      const { type, digits, terminator } = await prompt.waitForResult();
+      console.log(
+        "Prompted for digits, received digits",
+        type,
+        digits,
+        terminator
+      );
+
+      let e164number;
+      number = parsePhoneNumber("+" + digits);
```

Comment:
Is this getting you the expected response if the user inputs E164 format correctly with the + already included?

Reply:
I couldn't input + at all. Long pressing 0 has no effect. Thought it was a limitation of the platform?